### PR TITLE
Implement command unit test

### PR DESCRIPTION
**Pull request by FelixTing**
Resolve: #28 

Signed-off-by: Felix Ting <felix@iotechsys.com>

### CODE REVIEW

**Comment by FelixTing** [comment]
recheck

File: internal/handler/command_test.go [hunk 1]
```diff
@@ -1,36 +1,489 @@
...
+				if cv.Type != check {
+					t.Errorf("%s incorrect parsing. valueType: %s result: %v", tt.testName, tt.valueType, cv.Type)
+				}
+			}
+		})
+	}
+}
+
+func TestResourceOpSliceToMap(t *testing.T) {
+	ops := []contract.ResourceOperation{}
```
Comment (cloudxxx8) [inline internal/handler/command_test.go]: replace this line with
`var ops []contract.ResourceOperation`

File: internal/handler/command_test.go [hunk 2]
```diff
@@ -1,36 +1,489 @@
...
+		varsFindDeviceByValidId     = map[string]string{"id": "79ef304e-3b1e-4765-b509-798109c2467d", "command": "RandomValue_Int8"}
+		varsFindDeviceByInvalidId   = map[string]string{"id": "98621441-0f12-40ab-ad76-68edf1eeb4d5", "command": "RandomValue_Int8"}
+		varsFindDeviceByValidName   = map[string]string{"name": "Random-Integer-Generator01", "command": "RandomValue_Int8"}
+		varsFindDeviceByInvalidName = map[string]string{"name": "Random-Integer-Generator09", "command": "RandomValue_Int8"}
+		varsAdminStateLocked        = map[string]string{"name": "Random-Float-Generator01", "command": "RandomValue_Float32"}
+		varsProfileNotFound         = map[string]string{"name": "Random-Boolean-Generator01", "command": "error"}
+		varsCmdNotFound             = map[string]string{"name": "Random-Integer-Generator01", "command": "error"}
+		varsWriteInt8               = map[string]string{"name": "Random-Integer-Generator01", "command": "RandomValue_Int8"}
+	)
+	cache.Devices().UpdateAdminState("9a2b3ef2-6aed-41db-b868-79196c2a8b0a", contract.Locked)
```
Comment (cloudxxx8) [inline internal/handler/command_test.go]: make this id as a public constant in mock_deviceclient.go and handle the returned error variable from cache

File: internal/handler/command_test.go [hunk 3]
```diff
@@ -1,36 +1,489 @@
...
+	mock.ValidDeviceRandomBoolGenerator.Profile.Name = "error"
+	cache.Devices().Update(mock.ValidDeviceRandomBoolGenerator)
```
Comment (cloudxxx8) [inline internal/handler/command_test.go]: handle the returned error variable from cache

**Pull request merged**